x-pack/plugins/alerting/server/saved_objects/migrations.test.ts

```diff
@@ -1012,6 +1012,424 @@ describe('successful migrations', () => {
       });
     });
   });
+
+  describe('7.15.0', () => {
+    test('security solution is migrated to saved object references if it has 1 exceptionsList', () => {
+      const migration7150 = getMigrations(encryptedSavedObjectsSetup)['7.15.0'];
+      const alert = getMockData({
+        alertTypeId: 'siem.signals',
+        params: {
+          note: 'some note', // extra data to ensure we do not overwrite other params
+          exceptionsList: [
+            {
+              id: '123',
```

Comment:
Should the migration be updating the exceptionsList item so that the `id` is the `name` of the reference? The references array stores the name and id so that the id can be changed if necessary and the reference will still be valid. Right now, if the `id: 123` value is changed in the reference list, there is nothing linking that reference list item back to the exception list item. I think the exception list item should look like this after migration:

```
{
  id: `param:exceptionsList_0`,
  list_id: '456',
  type: 'detection',
  namespace_type: 'single',
}
```

Reply:
The `exceptionsList.id` is part of our user facing REST API and in part 1 ( #107064) we _always_ exchange it out for the references id array in the inject/extract as long as the references id exists. So technically although we still store the `exceptionsList.id` within the `alerting.params` it is no longer used except for the extreme case that we cannot find it in the `references` array. 

After this PR and migrations, the `references` array should always exist. We might eventually stop storing it altogether, but for now we are still storing it during an inject/extract.